akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -151,6 +155,8 @@ private[akka] object FanIn {
 
     def isCancelled(input: Int): Boolean = cancelled(input)
 
+    def isAllCompleted(): Boolean = (0 to inputCount - 1).map(completed).forall(_ == true)
```

Comment:
I see. 
Actually can also add test case where it's shown that subscription happening at the beginning for all sources even before pump starts working. Do you think it will be useful?